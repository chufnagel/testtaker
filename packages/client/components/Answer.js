import PropTypes from "prop-types";

const Answer = ({ id, text, onSelect }) => (
  <div
    className="radio-row"
    onClick={() => onSelect(id)}
    onKeyPress={() => onSelect(id)}
    aria-checked="false"
    role="radio"
    tabIndex="0"
  >
    <input key={id} name="answer" value={text} type="radio" />
    {`${id} - `} {`${text}`}
    <br />
  </div>
);




// const Answer = ({ id, text, onSelect }) => (
//     <input type="radio" onSelect={onSelect} id={id} value={text}> {text}<br /> </input>
// );

export default Answer;
