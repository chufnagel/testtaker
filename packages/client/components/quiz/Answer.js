import React, { memo } from "react";
import PropTypes from "prop-types";

const Answer = memo(({ questionId, choice, text, onSelect, key, checked }) => (
  <div
    className="radio-row"
    key={key}
    onClick={() => onSelect(questionId, choice)}
    onKeyPress={() => onSelect(questionId, choice)}
    aria-checked={checked}
    checked={checked}
    role="radio"
    tabIndex="0"
  >
    <input key={`${questionId}:${choice}`} name="answer" value={text} type="radio" />
    {`${choice} - `} {`${text}`}
    <br />
  </div>
));

Answer.propTypes = {
  questionId: PropTypes.number,
  choice: PropTypes.string,
  text: PropTypes.string,
  key: PropTypes.string,
  onSelect: PropTypes.func.isRequired,
  checked: PropTypes.bool,
};

Answer.defaultProps = {
  questionId: 1,
  choice: "A",
  key: "A",
  text: "Lorem ipsum",
  checked: false,
};

export default Answer;
