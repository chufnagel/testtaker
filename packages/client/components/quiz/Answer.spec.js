/* global beforeEach describe, expect, test */
import { shallow } from "enzyme";
import Answer from "./Answer";

// Answer.propTypes = {
//   questionId: PropTypes.string,
//   choice: PropTypes.string,
//   text: PropTypes.string,
//   onSelect: PropTypes.func.isRequired,
//   checked: PropTypes.bool,
// };

// Answer.defaultProps = {
//   questionId: 1,
//   choice: "A",
//   text: "Lorem ipsum",
//   checked: false,
// };

const mockFn = jest.fn();

describe("Answer", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Answer onSelect={mockFn} />);
  });
  test("It should render without crashing", () => {
    expect(wrapper.exists()).toEqual(true);
  });
  test("Answer matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
