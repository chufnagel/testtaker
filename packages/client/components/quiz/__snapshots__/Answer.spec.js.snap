// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer Answer matches snapshot 1`] = `
<div
  aria-checked={false}
  checked={false}
  className="radio-row"
  key="A"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="radio"
  tabIndex="0"
>
  <input
    key="1:A"
    name="answer"
    type="radio"
    value="Lorem ipsum"
  />
  A - 
   
  Lorem ipsum
  <br />
</div>
`;
